pkg/antctl/raw/multicluster/join.go

```diff
@@ -0,0 +1,282 @@
+	command := &cobra.Command{
+		Use:     "join",
+		Short:   "Join the ClusterSet from a member cluster",
+		Args:    cobra.MaximumNArgs(0),
+		Example: joinExamples,
+		RunE:    joinRunE,
+	}
+
+	o := joinOptions{}
+	joinOpts = &o
+	command.Flags().StringVarP(&joinOpts.LeaderClusterOpt.Namespace, "leader-namespace", "", "", "Namespace of the leader cluster")
+	command.Flags().StringVarP(&joinOpts.LeaderClusterOpt.ClusterID, "leader-cluster-id", "", "", "Cluster ID of the leader cluster")
+	command.Flags().StringVarP(&joinOpts.LeaderClusterOpt.Secret, "secret", "", "", "Secret of the leader cluster")
+	command.Flags().StringVarP(&joinOpts.LeaderClusterOpt.Server, "leader-server", "", "", "Server to the leader cluster")
+	command.Flags().StringVarP(&joinOpts.MemberClusterOpt.ClusterSet, "cluster-set", "", "", "ClusterSet to join")
+	command.Flags().StringVarP(&joinOpts.MemberClusterOpt.Namespace, "namespace", "", "", "Namespace of the member cluster")
+	command.Flags().StringVarP(&joinOpts.MemberClusterOpt.ClusterID, "cluster-id", "", "", "Cluster ID of the member cluster")
+	command.Flags().StringVarP(&joinOpts.ConfigFile, "config", "f", "", "Config file of the member cluster")
+
+	return command
+}
+
+func joinRunE(cmd *cobra.Command, args []string) error {
+	var err error
+	if err = joinOpts.validateAndComplete(); err != nil {
+		return err
+	}
+	k8sClient, err := common.NewClient(cmd)
+
+	memberClusterNamespace := joinOpts.MemberClusterOpt.Namespace
+	memberClusterID := joinOpts.MemberClusterOpt.ClusterID
+	memberClusterSet := joinOpts.MemberClusterOpt.ClusterSet
+	createdRes := []*unstructured.Unstructured{}
+	defer func() {
+		if err != nil {
+			fmt.Fprintf(cmd.OutOrStdout(), "Failed to join the ClusterSet, begin to delete the created resources.\n")
+			if err := common.Rollback(cmd, k8sClient, createdRes); err != nil {
+				fmt.Fprintf(cmd.OutOrStdout(), "Failed to rollback: %v\n", err)
+			}
+		}
+	}()
+	err = common.CreateClusterClaim(cmd, k8sClient, memberClusterNamespace, memberClusterSet, memberClusterID, &createdRes)
+	if err != nil {
+		return err
+	}
+	err = common.CreateClusterSet(cmd, k8sClient, memberClusterNamespace, memberClusterSet, joinOpts.LeaderClusterOpt.Server, joinOpts.LeaderClusterOpt.Secret,
+		joinOpts.MemberClusterOpt.ClusterID, joinOpts.LeaderClusterOpt.ClusterID, joinOpts.LeaderClusterOpt.Namespace, &createdRes)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(cmd.OutOrStdout(), "Waiting for member cluster ready\n")
+	if err = waitForMemberClusterReady(cmd, k8sClient); err != nil {
+		fmt.Fprintf(cmd.OutOrStdout(), "Failed to wait for member cluster ready: %v\n", err)
+		return err
+	}
+	fmt.Fprintf(cmd.OutOrStdout(), "Member cluster joined successfully\n")
+
+	return nil
+}
+
+func waitForMemberClusterReady(cmd *cobra.Command, k8sClient client.Client) error {
+	fmt.Fprintf(cmd.OutOrStdout(), "Waiting for ClusterSet \"%s\" in Namespace %s to be registered in leader cluster\n",
```

Comment:
Probably remove "in Namespace %s". I feel the information is not useful to users.